### PR TITLE
[pangolin] Update to 0.8.

### PR DESCRIPTION
**Pull request by Sibras**
Updates to latest tag version 0.8.

Multiple features were removed as the newer version has moved them from being selectable to always being built. As such eigen3 is now a hard dependency and no longer selectable as part of feature dependencies.

This is required to support using newer versions of ffmpeg (5.0+)

See PR [#23312](https://github.com/microsoft/vcpkg/pull/23312)

### CODE REVIEW

File: ports/pangolin/portfile.cmake [hunk 1]
```diff
@@ -6,28 +6,19 @@ endif()
...
-    REF dd801d244db3a8e27b7fe8020cd751404aa818fd #v0.6
-    SHA512 8004ab6f146f319df41e4b8d4bdb6677b8faf6db725e34fea76fcbf065522fa286d334c2426dcb39faf0cfb3332946104f78393d2b2b2418fe02d91450916e78
+    REF v0.8
+    SHA512 d4ca405097e8c439a4f74495f374bc5d5e4febafcf59ee88d985a8764ed36da1753ca4a3a73476dfb74c7d92df31a99242df6e1b47c648e860eee835a6f4f434
     HEAD_REF master
-    PATCHES
-        fix-includepath-error.patch # include path has one more ../
-        fix-cmake-version.patch
-        fix-build-error-in-vs2019.patch
-        fix-dependencies.patch
```
Comment (JonLiu1993) [inline ports/pangolin/portfile.cmake:16]: @Sibras, thanks for the pr, we should also delete the patch files in the folder if the patches are no longer needed.

File: ports/pangolin/portfile.cmake [hunk 2]
```diff
@@ -85,11 +70,6 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
...
 endif()
 
-if(VCPKG_TARGET_IS_WINDOWS)
-    # Copy missing header file
-    file(COPY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/src/include/pangolin/pangolin_export.h" DESTINATION "${CURRENT_PACKAGES_DIR}/include/pangolin")
-endif()
-
 # Put the license file where vcpkg expects it
 file(COPY "${CURRENT_PORT_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 file(INSTALL "${SOURCE_PATH}/LICENCE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (JonLiu1993) [inline ports/pangolin/portfile.cmake:75]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENCE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

```

Comment (JonLiu1993) [reply ports/pangolin/portfile.cmake:75]: Add a blank line at the end to ensure that it is consistent with the previous format.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/pangolin/portfile.cmake (hunk 2 above)
Comment (Sibras) [reply ports/pangolin/portfile.cmake:75]: The previous format didnt have a newline according to git diff

Comment (quyykk) [inline ports/pangolin/portfile.cmake:75]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
@Sibras, when I tested the features by command "./vcpkg install pangolin[*]:x86-windows" I got this error, please take a look:
````
Could not find video in pangolin.
note: updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.
````

**Comment by Sibras** [comment]
> @Sibras, when I tested the features by command "./vcpkg install pangolin[*]:x86-windows" I got this error, please take a look:
> 
> ```
> Could not find video in pangolin.
> note: updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.
> ```

There was an issue with tools feature that is now fixed.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Expect feature ''uvc", all features are tested successfully in the following triplet:
- x86-windows
- x64-windows

**Review by dg0yt — CHANGES_REQUESTED** [review]
This PR removes features. It can break upgrades for users which depend on these features. Alternative:
Keep the features as no-ops.

**Comment by Sibras** [comment]
> This PR removes features. It can break upgrades for users which depend on these features. Alternative: Keep the features as no-ops.

OK Ive added the removed features back to the vcpkg.json file

**Comment by JonLiu1993** [comment]
> pangolin

@Sibras, If you restore that code, we will also encounter the same problem before, the `vedio` feature cannot be tested

**Comment by Sibras** [comment]
OK so should I keep the features as no-ops as suggested or remove them entirely as initially done?

**Comment by dg0yt** [comment]
> > > when I tested the features by command "./vcpkg install pangolin[*]:x86-windows" I got this error, please take a look:
> > > 
> > > ```
> > > Could not find video in pangolin.
> > > note: updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.
> > > ```
> >
> > If you restore that code, we will also encounter the same problem before, the `vedio` feature cannot be tested

No.

> should I keep the features as no-ops as suggested or remove them entirely as initially done?

You must keep the no-op features. 

Ironically, this problem was actually caused by the removal of feature `video` while it was still used as a dependency. 
User may have a dependency on `pangolin[video]`, so will get that problem when the feature is not kept.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/pangolin/vcpkg.json [hunk 3]
```diff
@@ -18,20 +18,13 @@
...
-    "video"
+    "png"
   ],
   "features": {
     "eigen": {
-      "description": "Build support for Eigen matrix types",
-      "dependencies": [
-        "eigen3"
-      ]
+      "description": "Unused option left for backwards compatibility with older versions"
```
Comment (vicroms) [inline ports/pangolin/vcpkg.json:27]: It should be OK to remove these features entirely.

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/pangolin/vcpkg.json (hunk 3 above)
Comment (dg0yt) [reply ports/pangolin/vcpkg.json:27]: And break `vcpkg upgrade`?

Comment (vicroms) [reply ports/pangolin/vcpkg.json:27]: You're right, I forgot about `vcpkg upgrade`... @ras0219-msft and @BillyONeal what do you think?

Comment (dg0yt) [reply ports/pangolin/vcpkg.json:27]: Note the previous comments in this PR ...

**Review by vicroms — APPROVED** [review]

**Pull request merged**